### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by kj7rrv** [florisboard/florisboard#619]
I was just typing a DDG search in Fennec F-Droid's address bar.
~~~ 1618203425190.stacktrace ~~~

java.lang.OutOfMemoryError: Failed to allocate a 524304 byte allocation with 394056 free bytes and 384KB until OOM, target footprint 201326592, growth limit 201326592
	at java.util.HashMap.resize(HashMap.java:703)
	at java.util.HashMap.putVal(HashMap.java:662)
	at java.util.HashMap.put(HashMap.java:611)
	at org.json.JSONObject.put(JSONObject.java:272)
	at org.json.JSONTokener.readObject(JSONTokener.java:394)
	at org.json.JSONTokener.nextValue(JSONTokener.java:104)
	at org.json.JSONObject.<init>(JSONObject.java:164)
	at org.json.JSONObject.<init>(JSONObject.java:181)
	at dev.patrickgold.florisboard.ime.text.gestures.GlideTypingManager$setWordData$1.invokeSuspend(GlideTypingManager.kt:79)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)

**Comment by Glitchy-Tozier** [comment]
1. I like your browser & search engine choice!  :)
2. This might be a duplicate of #569

**Comment by patrickgold** [comment]
Thanks for your bug report! So the crash occurred within the GlideTypingManager and this particular memory bug should be fixed in #625. Closing this issue thus. Memory crashes in general are though still very likely to occur, just not through this specific line.

**Comment by kj7rrv** [comment]
Okay, thanks! I do have glide typing on, but IIRC I wasn't using it at the time this happened.On Apr 12, 2021 10:06, Patrick Goldinger ***@***.***> wrote:
Thanks for your bug report! So the crash occurred within the GlideTypingManager and this particular memory bug should be fixed in #625. Closing this issue thus. Memory crashes in general are though still very likely to occur, just not through this specific line.
